File: frontend-server/src/app/model/dto/united/admin-common-data-dto.ts
import { DepartmentDayType } from "../../department-day-type";
import { BasicDTO } from "../basic-dto";
import { ShiftPattern } from "../../shift-pattern";
import { PatternUnit } from "../../pattern-unit";
import { RatioColumn } from "../../ratio-column";

export class AdminCommonDataDTO {
  departmentDayTypes: DepartmentDayType[];
  patternDTOs:        BasicDTO<ShiftPattern, PatternUnit>[];
  ratioColumns:       RatioColumn[];
}

File: work-schedule-service/src/main/java/com/averude/uksatse/scheduler/microservice/workschedule/model/AdminCommonDataDTO.java
package com.averude.uksatse.scheduler.microservice.workschedule.model;

import com.averude.uksatse.scheduler.core.model.dto.BasicDto;
import com.averude.uksatse.scheduler.core.model.entity.DepartmentDayType;
import com.averude.uksatse.scheduler.core.model.entity.PatternUnit;
import com.averude.uksatse.scheduler.core.model.entity.ShiftPattern;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@RequiredArgsConstructor
public class AdminCommonDataDTO {
    private final List<DepartmentDayType> departmentDayTypes;
    private final List<? extends BasicDto<ShiftPattern, PatternUnit>> patternDTOs;
}
